From 02ffc5158d1ad270e0b5c7ce6dfe4414a6ec029f Mon Sep 17 00:00:00 2001
From: Michal Domonkos <mdomonko@redhat.com>
Date: Wed, 31 Jul 2024 16:19:40 +0200
Subject: [PATCH] Fix division by zero in elfdeps (RhBug:2299414)

Assume that the section does not hold a table if sh_entsize is 0 (as
specified in the elf(5) man page) and just skip it if that's the case.
---
 tools/elfdeps.c | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/tools/elfdeps.c b/tools/elfdeps.c
index cb388f08d..822359ab9 100644
--- a/tools/elfdeps.c
+++ b/tools/elfdeps.c
@@ -196,6 +196,8 @@ static void processVerNeed(Elf_Scn *scn, GElf_Shdr *shdr, elfInfo *ei)
 static void processDynamic(Elf_Scn *scn, GElf_Shdr *shdr, elfInfo *ei)
 {
     Elf_Data *data = NULL;
+    if (shdr->sh_entsize == 0)
+	return;
     while ((data = elf_getdata(scn, data)) != NULL) {
 	for (int i = 0; i < (shdr->sh_size / shdr->sh_entsize); i++) {
 	    const char *s = NULL;
-- 
2.45.2

